pilot/pkg/model/config.go

```diff
@@ -889,8 +889,18 @@ func (store *istioConfigStore) AuthenticationPolicyByDestination(service *Servic
 		matchLevel := 0
 		if len(policy.Targets) > 0 {
 			for _, dest := range policy.Targets {
-				if service.Hostname != ResolveShortnameToFQDN(dest.Name, spec.ConfigMeta) {
-					continue
+				// When labels is specified, use labels to match the policy. Otherwise, fallback to use host name.
+				if len(dest.Labels) != 0 {
+					log.Debugf("found label selector on auth policy (%s/%s): %s", dest.Labels, spec.Namespace, spec.Name)
+					destLabels := Labels(dest.Labels)
+					if !destLabels.SubsetOf(labels) {
```

Comment:
is it possible several policies pass this check ? if yes, seems current logic is randomly select one ? Is this expected or shall we select policy which has `more matching` labels than others ?

Reply:
It's fine for now. The current spec cannot solve this fully regardless. While give preference to `more matching` can help in some cases, but more confusing in general, imo. For example, which selector is more specific: 'app:foo' vs 'env:prod' ? We will need to solve this in the new API, so for now, it's ok to keep this part simple.

Reply:
Right, the current logic selects a random one if multiple policies matched in this level, the logic does the same thing in this case.
I don't understand what does `more matching` mean actually, it seems we only have either `matched` or `not matched`?

Reply:
For example - 
policy1 labels: "foo":"1", "bar":"2"
policy2 labels: "foo": "1"
policy3 labels: "bar": "2", "ooo":"3"
service labels: "foo":"1", "bar":"2", "ooo":"3"

so policy1 and policy3 are `more matching` than policy2; 
current logic may select policy2, probably select(randomly) from policy1 or policy3 is better ?

Reply:
Thanks for the clarification, I think we may have a better way to solve this in the new API, I'll keep it as-is for now, we can have some more discussions before making the API public (currently the labels are hidden in the API).